torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -196,14 +213,16 @@ def speculate_subgraph(
     source_target=None,
     always_restore=False,
     enable_grad=None,
-    # NOTE [Temporary argument `manually_set_subgraph_inputs`]
-    # If manually_set_subgraph_inputs=True, then we manually add
-    # the `sub_args` to `subgraph`, if False then we rely
-    # on tracer's lifting mechanism to lift these args.
+    # NOTE [Temporary argument `set_subgraph_inputs`]
+    # If set_subgraph_inputs="manual", we manually add the `sub_args` to `subgraph`,
+    # If "automatic", we rely on tracer's lifting mechanism to lift these args.
+    # If "flatten_manual", we first flatten the sub_args into a flattend list, then manually
+    # set the flattend inputs to the subgraph. This saves us from having to mapping the automatically
+    # lifted inputs back to their original position, where ordering of input matters e.g. map.
```

Comment:
This arg isn't temporary anymore. Let's delete the next 3 lines ("Default `True` is temporary").


We should say something like
- if your HOP supports arbitrary inputs, use set_subtraph_inputs="automatic"
- if your HOP supports Tensor and symnode inputs, use set_subgraph_inputs="flatten_manual"

Reply:
For "manual" option, the arg can actually be AutogradFunctionContextVariable or it supports maybe_fx_node(), maybe we can change the recommendation to "This is the most restrictive option (your function must only accept Tensor and symnode inputs or you know exactly how to create a proxy for it) and we do not recommend it."

Reply:
Yeah go for it